liiatools_pipeline/sensors/job_success_sensor.py

```diff
@@ -19,19 +19,12 @@
 from liiatools_pipeline.ops.common_config import CleanConfig
 
 
-def find_previous_matching_dataset_run(run_records, dataset, context):
-    try:
-        previous_run_id = [
-            run.dagster_run.run_id
-            for run in run_records
-            if dataset == run.dagster_run.tags["dataset"]
-        ]
-
-    except KeyError:
-        context.log.error(
-            f"'dataset' not found in tags. No previous run id found"
-        )
-        previous_run_id = None
```

Comment:
Any reason for removing this error handling?  There's still a small chance the dataset tag may not exist in a previous run, so it's better to have protections in place for this possibility.

Reply:
This was removed because we now filter for the "dataset" tag (see line 45 as an example) so as long as the .get_run_records function is working then there will always be a "dataset" tag (or no runs at all which also fine)